src/test/java/com/amazon/opendistroforelasticsearch/security/IndexIntegrationTests.java

```diff
@@ -473,7 +473,13 @@ public void testCCSIndexResolve() throws Exception {
 
         res = rh.executeGetRequest("/*:.abc-6,.abc-6/_search", encodeBasicHeader("ccsresolv", "nagilum"));
         Assert.assertEquals(HttpStatus.SC_OK, res.getStatusCode());
+
         //TODO: Change for 25.0 to be forbidden (possible bug in ES regarding ccs wildcard)
+
+        // invalid_index_name_exception should be thrown and responded when invalid index name is mentioned in requests.
+        res = rh.executeGetRequest("_pdt_data/_search", encodeBasicHeader("ccsresolv", "nagilum"));
```

Comment:
Can you add a test for the same index name in ticket `_##pdt_data` ?

Reply:
I've tried that, but the `#` is identified as a fragment and caused  `java.lang.IllegalArgumentException: Illegal character in fragment at index`.

Reply:
Was this not the original intention of the ticket? `_pdt_data` is missing index which is already correctly throwing index not found exception?

Reply:
I agree that this it exactly the index name mentioned in the ticket. But `##` causes `java.net.URISyntaxException`at `at java.base/java.net.URI.create(URI.java:906)`. I hope [this post](https://stackoverflow.com/questions/20941947/java-net-uri-illegal-character-in-fragment-at-index-xxx-due-to-character) helps explain the reason. 
I've changed the index name in the test and added a `#`.

Reply:
Please check ES behavior without security plugin being installed.

Reply:
Without security plugin being installed, ES behaves in the way that we expect after this fix. It responds a invalid_index_name_exception to `curl -XGET 'http://localhost:9200/_##pdt_data/_search'`.

Reply:
Is it possible to add this in the tests?

Reply:
@sujithvm As I mentioned in the previous comments, using `##` fails the unit test due to `java.lang.IllegalArgumentException: Illegal character in fragment at index`, which is not related to ES or security plugin.

Reply:
Applied url encoding to the test. Thanks @sujithvm!